Feature/155186331 read cluster files

alfonsomunozpomer opened the pull request:
Companion PR of https://github.com/ebi-gene-expression-group/atlas/pull/83. The branch over there has a version of this branch bundled if you want to test it.

alfonsomunozpomer committed:
Change default example to our new SC poster child experiment

alfonsomunozpomer committed:
Make separate requests for gene expression and cell clusters

alfonsomunozpomer committed:
Remove legend and set the same symbol shape for all series in both charts; add a wider colour palette for clusters

monicajianu commented on src/ClusterTSnePlot.js:
```diff
@@ -30,34 +30,53 @@ const ClusterTSnePlot = (props) => {
       text: `Clusters`
+    },
+    legend: {
+      enabled: false
```
I think we'll need to find a way to bring back the legend in future iterations... It will probably get very confusing once we add clustering by metadata.

monicajianu commented on src/TSnePlotView.js:
```diff
@@ -16,97 +16,108 @@ class ExperimentPageView extends React.Component {
   constructor(props) {
```
Shouldn't this class be named `TSnePlotView` to match the name of the file?

monicajianu commented on src/TSnePlotView.js:
```diff
@@ -16,97 +16,108 @@ class ExperimentPageView extends React.Component {
-                           plotData={data}
+                           plotData={cellClustersData}
                            perplexities={perplexities}
                            perplexity={perplexity}
```
Would it be better if `perplexity` was renamed to something like `selectedPerplexity`? The same could be said about `k`.

alfonsomunozpomer replied:
Makes sense.

monicajianu commented on src/ClusterTSnePlot.js:
```diff
@@ -30,34 +30,53 @@ const ClusterTSnePlot = (props) => {
       text: `Clusters`
+    },
+    legend: {
+      enabled: false
```
I think we'll need to find a way to bring back the legend in future iterations... It will probably get very confusing once we add clustering by metadata.

alfonsomunozpomer replied:
I’d rather do a separate legend component, similar to what we have in the [multiexperiment heatmap](https://www.ebi.ac.uk/gxa/genes/ensg00000172023). At one point, before we implemented the *Filters* button, it was interactive and could hide/display series in Highcharts as well. I don’t know if this would be a requirement. The trouble I have with Highcharts legend is that it takes space from the chart plot rectangle, squashing it vertically, and it doesn’t align nicely with the right plot.

monicajianu commented on src/TSnePlotView.js:
```diff
@@ -16,97 +16,108 @@ class ExperimentPageView extends React.Component {
   constructor(props) {
```
Shouldn't this class be named TSnePlotView to match the name of the file?

alfonsomunozpomer replied:
Good catch.

monicajianu reviewed (changes requested):
Tiny, tiny comments, everything else looks good.

alfonsomunozpomer committed:
Change name of component to match name of file
https://github.com/ebi-gene-expression-group/scxa-experiment-page-tsne-plot-view/pull/4

alfonsomunozpomer committed:
Renamed parameters to be more explicit about their purpose
https://github.com/ebi-gene-expression-group/scxa-experiment-page-tsne-plot-view/pull/4

alfonsomunozpomer committed:
Fix tests after change of parameter

alfonsomunozpomer committed:
Merge branch 'master' into feature/155186331-read-cluster-files
